core/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -233,15 +234,34 @@ private void applyDeletedIndices(final ClusterChangedEvent event) {
             if (idxService != null) {
                 indexSettings = idxService.getIndexSettings();
                 deleteIndex(index, "index no longer part of the metadata");
-            } else {
-                final IndexMetaData metaData = previousState.metaData().getIndexSafe(index);
+            } else if (previousState.metaData().hasIndex(index.getName())) {
+                // The deleted index was part of the previous cluster state, but not loaded on the local node
+                final IndexMetaData metaData = previousState.metaData().index(index);
                 indexSettings = new IndexSettings(metaData, settings);
-                indicesService.deleteUnassignedIndex("closed index no longer part of the metadata", metaData, event.state());
+                indicesService.deleteUnassignedIndex("deleted index was not assigned to local node", metaData, event.state());
+            } else {
+                // The previous cluster state's metadata also does not contain the index,
+                // which is what happens on node startup when an index was deleted while the
+                // node was not part of the cluster.  In this case, try reading the index
+                // metadata from disk.  If its not there, there is nothing to delete.
+                // First, though, verify the precondition for applying this case by
+                // asserting that the previous cluster state is not initialized/recovered.
+                assert previousState.blocks().hasGlobalBlock(GatewayService.STATE_NOT_RECOVERED_BLOCK);
+                final IndexMetaData metaData = indicesService.verifyIndexIsDeleted(index, event.state());
+                if (metaData != null) {
+                    indexSettings = new IndexSettings(metaData, settings);
+                } else {
+                    indexSettings = null;
+                }
             }
-            try {
-                nodeIndexDeletedAction.nodeIndexDeleted(event.state(), index, indexSettings, localNodeId);
-            } catch (Throwable e) {
-                logger.debug("failed to send to master index {} deleted event", e, index);
+            // indexSettings can only be null if there was no IndexService and no metadata existed
+            // on disk for this index, so it won't need to go through the node deleted action anyway
+            if (indexSettings != null) {
+                try {
+                    nodeIndexDeletedAction.nodeIndexDeleted(event.state(), index, indexSettings, localNodeId);
+                } catch (Throwable e) {
```

Comment:
Can you make this `Exception` instead of `Throwable`? this is a bad place to swallow `OutOfMemoryError` since people may not have debug logging turned on